modules/dop/component-protocol/components/project-pipeline/myPage/tabsTable/filterContainer/customFilter/condition.go

```diff
@@ -135,3 +142,27 @@ func (p *CustomFilter) AppCondition() (*model.SelectCondition, error) {
 	condition.ConditionBase.Placeholder = cputil.I18n(p.sdk.Ctx, "please-choose-application")
 	return condition, nil
 }
+
+func (p *CustomFilter) BranchCondition() (*model.SelectCondition, error) {
+	branches, err := p.ProjectPipelineSvc.ListRef(p.sdk.Ctx, deftype.ProjectPipelineRefList{
+		ProjectID:    p.InParams.ProjectID,
+		IdentityInfo: apistructs.IdentityInfo{UserID: p.sdk.Identity.UserID},
+	})
+	if err != nil {
+		return nil, err
+	}
+	cond := model.NewSelectCondition("branch", cputil.I18n(p.sdk.Ctx, "branch"), func() []model.SelectOption {
+		selectOptions := make([]model.SelectOption, 0, len(branches))
+		for _, v := range branches {
+			selectOptions = append(selectOptions, *model.NewSelectOption(func() string {
+				return v
```

Comment:
why not just `*model.NewSelectOption(v, ...` ?

Reply:
done